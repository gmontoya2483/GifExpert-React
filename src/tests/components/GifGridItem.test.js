import {shallow} from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas en componente <GifGridItem/>', ()=> {

    const title = 'Un título'
    const url = 'https://localhost/algo.jpg'
    let wrapper = shallow(<GifGridItem title={title} url={url}/>);

    beforeEach(()=>{

    })

    test('Debe de mostrar <GifGridItem/> corectamente (hacer match con el snapshot)', ()=>{

        expect( wrapper ).toMatchSnapshot();
    });


    test('debe de tener un parrafo con el título', ()=> {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe de tener la imagen igual al url y alt de los props', ()=> {

        const img = wrapper.find('img');
        // console.log( img.prop('src') );

        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );

    })

    test('debe de tener la clase animate__fadeInt', ()=>{
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);

    })



})
