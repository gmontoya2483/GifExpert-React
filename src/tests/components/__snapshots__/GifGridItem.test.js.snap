// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem/> Debe de mostrar <GifGridItem/> corectamente (hacer match con el snapshot) 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
